### PR TITLE
Misc tidying

### PR DESCRIPTION
**Pull request by pmarsh-scottlogic**
## Description

Some low-hanging fruit tidying that I did while scouring the codebase in order to write docs.

## Notes

- removes a bunch of redundant comments
- renames `queryPromptEvaluationModel` to `evaluatePrompt`. It now returns a boolean instead of wrapping it in an object for no good reason.
- `queryDocuments` now just returns a string rather than wrapping it in a object for no good reason.
- A bunch of minor local code changes not worth mentioning explicitly.
- Moves handbook page components into their own folder. Moves _Attacks.ts_ and _Glossary.ts_ to the handbook pages folder.

## Checklist

Have you done the following?

- [ ] Linked the relevant Issue - No issue
- [ ] Added tests - no added functionality so no additional tests
- [x] Ensured the workflow steps are passing

### CODE REVIEW

File: backend/src/controller/chatController.ts [hunk 1]
```diff
@@ -283,15 +282,18 @@ async function handleChatToGPT(req: OpenAiChatRequest, res: Response) {
 	}
 
 	let updatedChatHistory = levelResult.chatHistory;
-	totalSentEmails.push(...levelResult.chatResponse.sentEmails);
+
+	const totalSentEmails: EmailInfo[] = [
```
Comment (pmarsh-scottlogic) [inline backend/src/controller/chatController.ts:288]: before, this was being instantiated then mutated once to add on the new emails sent as a result of the message. Now we just instantiate it including the new emails

File: backend/src/langchain.ts [hunk 2]
```diff
@@ -79,87 +75,75 @@ function initPromptEvaluationModel(configPromptEvaluationPrompt: string) {
...
 			query: question,
 		})) as QaChainReply;
-		// log the time taken
-		console.debug(`QA model call took ${Date.now() - startTime}ms`);
 
+		console.debug(`QA model call took ${Date.now() - startTime}ms`);
 		console.debug(`QA model response: ${response.text}`);
-		const result: ChatAnswer = {
-			reply: response.text,
-			questionAnswered: true,
```
Comment (pmarsh-scottlogic) [inline backend/src/langchain.ts:113]: We weren't using this `questionAnswered` value anywhere.

File: backend/src/openai.ts [hunk 3]
```diff
@@ -152,14 +152,10 @@ async function handleAskQuestionFunction(
...
 			: '';
-		return {
-			reply: (
-				await queryDocuments(params.question, configQAPrompt, currentLevel)
-			).reply,
-		};
+		return await queryDocuments(params.question, configQAPrompt, currentLevel);
 	} else {
 		console.error('No arguments provided to askQuestion function');
-		return { reply: "Reply with 'I don't know what to ask'" };
```
Comment (pmarsh-scottlogic) [inline backend/src/openai.ts:162]: there was no reason to wrap this in an object either

**Review by AAloshine-scottlogic — APPROVED** [review]
looks good to me but you may want someone else whos more familiar with the codebase to also review!

**Comment by pmarsh-scottlogic** [comment]
Thanks @AAloshine-scottlogic I'll leave the PR up until end of Monday, say, so folk can weigh in if they want to

File: backend/src/openai.ts [hunk 4]
```diff
@@ -371,20 +364,13 @@ async function getFinalReplyAfterAllToolCalls(
...
-	const openai = getOpenAI();
 	let gptReply: ChatGptReply | null = null;
 
 	do {
-		gptReply = await chatGptChatCompletion(
-			updatedChatHistory,
-			chatModel,
-			openai
-		);
+		gptReply = await chatGptChatCompletion(updatedChatHistory, chatModel);
```
Comment (chriswilty) [inline backend/src/openai.ts:370]: This is inside a do..while loop, so every time it is invoked, we will be creating a new OpenAI instance inside the chatGptChatCompletion function.

File: frontend/src/components/HandbookOverlay/Pages/HandbookAttacks.test.tsx [hunk 5]
```diff
@@ -1,7 +1,11 @@
 import { render, screen } from '@testing-library/react';
 import { describe, expect, test } from 'vitest';
 
-import { ATTACKS_LEVEL_2, ATTACKS_LEVEL_3, ATTACKS_ALL } from '@src/Attacks';
+import {
+	ATTACKS_LEVEL_2,
+	ATTACKS_LEVEL_3,
+	ATTACKS_ALL,
+} from '@src/components/HandbookOverlay/Pages/Attacks';
```
Comment (chriswilty) [inline frontend/src/components/HandbookOverlay/Pages/HandbookAttacks.test.tsx:8]: This is now in same directory, so can just use `from './Attacks'`

File: frontend/src/components/HandbookOverlay/Pages/HandbookAttacks.tsx [hunk 6]
```diff
@@ -1,4 +1,8 @@
-import { ATTACKS_ALL, ATTACKS_LEVEL_2, ATTACKS_LEVEL_3 } from '@src/Attacks';
+import {
+	ATTACKS_ALL,
+	ATTACKS_LEVEL_2,
+	ATTACKS_LEVEL_3,
+} from '@src/components/HandbookOverlay/Pages/Attacks';
```
Comment (chriswilty) [inline frontend/src/components/HandbookOverlay/Pages/HandbookAttacks.tsx:5]: Same here: `from './Attacks'`

File: frontend/src/components/HandbookOverlay/Pages/HandbookGlossary.test.tsx [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import { render, screen } from '@testing-library/react';
 import { describe, expect, test } from 'vitest';
 
-import { GLOSSARY } from '@src/Glossary';
+import { GLOSSARY } from '@src/components/HandbookOverlay/Pages/Glossary';
```
Comment (chriswilty) [inline frontend/src/components/HandbookOverlay/Pages/HandbookGlossary.test.tsx:4]: Same idea here - this file is now in same directory.

File: frontend/src/components/HandbookOverlay/Pages/HandbookGlossary.tsx [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import { GLOSSARY } from '@src/Glossary';
+import { GLOSSARY } from '@src/components/HandbookOverlay/Pages/Glossary';
```
Comment (chriswilty) [inline frontend/src/components/HandbookOverlay/Pages/HandbookGlossary.tsx:1]: And same here - this file is now in same directory.

**Review by chriswilty — CHANGES_REQUESTED** [review]
Worthwhile cleanup 👍

Just a handful of things spotted.

File: backend/src/openai.ts [hunk 9]
```diff
@@ -293,11 +287,10 @@ async function chatGptChatCompletion(
 	}
 }
 
-function getChatCompletionsFromHistory(
+function getChatCompletionsInLimitedContextWindow(
```
[the remark this answers is not in the document]
Comment (pmarsh-scottlogic) [reply backend/src/openai.ts:290]: I'ts a description I got from the console.log in the same function.

A quick Google reveals that it is a recognised term [[article](https://www.hopsworks.ai/dictionary/context-window-for-llms), [article](https://blog.google/technology/ai/long-context-window-ai-models/)].

It is also a tem userd by OpenAI, e.g. https://platform.openai.com/docs/models/gpt-4-and-gpt-4-turbo

File: backend/src/openai.ts [hunk 10]
```diff
@@ -293,11 +287,10 @@ async function chatGptChatCompletion(
...
 	chatHistory: ChatMessage[],
 	gptModel: CHAT_MODELS
 ): ChatCompletionMessageParam[] {
-	// take only completions to send to model
 	const completions = chatHistory.reduce<ChatCompletionMessageParam[]>(
```
[the remark this answers is not in the document]
Comment (pmarsh-scottlogic) [reply backend/src/openai.ts:294]: yeah good spot, I'll change that

[the remark this answers is not in the document]
Comment (pmarsh-scottlogic) [reply backend/src/openai.ts:294]: We need to get all the completions, but not every chatMessage has a completion. So i've swapped it for a map then a filter:

```typescript
const completions = chatHistory
	.map((chatMessage) =>
		'completion' in chatMessage ? chatMessage.completion : null
	)
	.filter(
		(completion) => completion !== null
	) as ChatCompletionMessageParam[];
```

I'm sure there's a nicer way though. I'm sure you've helped me with this particular scenario before as well, but I can't find where

[the remark this answers is not in the document]
Comment (chriswilty) [reply backend/src/openai.ts]: Oh, this rings a bell now... It was the awkward typings that meant the reduce seemed like the least verbose solution. I seem to recall you couldn't simply filter then map, as then you needed the type check in both places. Oh well.

File: backend/src/utils/token.ts [hunk 11]
```diff
@@ -7,6 +7,7 @@ import { promptTokensEstimate, stringTokens } from 'openai-chat-tokens';
 import { CHAT_MODELS } from '@src/models/chat';
 import { chatGptTools } from '@src/openai';
 
+// The size of each model's context window in muber of tokens. https://platform.openai.com/docs/models
```
Comment (chriswilty) [inline backend/src/utils/token.ts:10]: muber 🤪

**Review by chriswilty — APPROVED** [review]
All good 👌

**Pull request merged**